internal/service/service_test.go

```diff
@@ -0,0 +1,119 @@
+package service_test
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/andreipimenov/golang-training-2021/internal/mock"
+	"github.com/andreipimenov/golang-training-2021/internal/model"
+	"github.com/andreipimenov/golang-training-2021/internal/service"
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/suite"
+)
+
+const (
+	invalidTicker  = "ABCDEFG"
+	validTicker    = "AAPL"
+	date           = "2021-07-26"
+	apiKey         = "1234"
+	invalidApiKey  = ""
+	invalidDate    = "2089-12-21"
+	badURL         = "abcdefg"
+	stockAPIFormat = "https://www.alphavantage.co/query?function=TIME_SERIES_DAILY&outputsize=full&symbol=%s&apikey=%s"
+)
+
+type serviceTestSuite struct {
+	suite.Suite
+	clientMock *mock.HTTPClient
+	repoMock   *mock.Repository
+	service    *service.Service
+}
+
+func (suite *serviceTestSuite) SetupTest() {
+	client := &mock.HTTPClient{}
+	repo := &mock.Repository{}
+	s := service.New(&zerolog.Logger{}, repo, apiKey)
```

Comment:
The same issue as here: https://github.com/andreipimenov/golang-training-2021/pull/82#discussion_r689499835
and here: https://github.com/andreipimenov/golang-training-2021/pull/79#discussion_r694665193 . Please cooperate with guys and solve it.